cli/tests/activate.bats

```diff
@@ -4135,3 +4147,169 @@ EOF
 
   zsh_repeat_activation_PATH .zshenv.extra .zshrc
 }
+
+# This test ensures forwards-compatible changes to state in:
+#
+# 1. the structure of `activations.json`
+# 2. arguments and environment variables used in activation scripts
+attach_previous_release() {
+  shell="${1?}"
+  mode="${2?}"
+
+  echo "$HOOK_ONLY_ONCE" | "$FLOX_BIN" edit -f -
+
+  export FLOX_SHELL="$shell"
+  case "$shell" in
+    bash)
+      rc_file=".bashrc";;
+    fish)
+      rc_file="config.fish";;
+    tcsh)
+      rc_file=".tcshrc";;
+    zsh)
+      rc_file=".zshenv";;
+    *)
+      echo "Unsupported shell: $shell"
+      exit 1
+      ;;
+  esac
+
+  expected_content="Sourcing ${rc_file}
+Setting PATH from ${rc_file}"
+
+  mkfifo activate_finished
+  # Will get cat'ed in teardown
+  TEARDOWN_FIFO="$PROJECT_DIR/teardown_activate"
+  mkfifo "$TEARDOWN_FIFO"
+
+  # Pre-fetch before doing an activation.
+  nix run "github:flox/flox/${GIT_LATEST_TAG}" -- --version
+
+  # Start an activation with the previously released version.
+  # Our tcsh quoting appears to be broken so don't quote $TEARDOWN_FIFO
+  nix run "github:flox/flox/${GIT_LATEST_TAG}" -- \
+    activate -- \
+    "$(which "$shell")" -c "echo > activate_finished && echo > $TEARDOWN_FIFO" > output 2>&1 &
+
+  timeout 15s cat activate_finished || echo "activation didn't complete in time" >&2
+  run cat output
+  # This the only place we use `--partial` because we only want to know that the
+  # intial activation started.
+  assert_output --partial - << EOF
+sourcing hook.on-activate for first time
+${expected_content}
+EOF
+
+  # Attach to the activation with the current version and then nest another
+  # attachment just to be sure. All assertions are on complete output (no
+  # partial) so that we can detect errors.
+  case "$mode" in
+    interactive)
+      interactive_command="$FLOX_BIN activate -- true"
```

Comment:
suggestion blocking: we currently unset FLOX_SHELL as part of activation so I think you need to set it again or rebase on FLOX_SHELL_NOCLOBBER